slot.go

```diff
@@ -134,30 +176,75 @@ func (s *Slot) findKey(search []*pkcs11.Attribute) (*Key, error) {
 func describeKey(obj obj) (c, t string, err error) {
 	attrs, err := obj.ctx.GetAttributeValue(obj.h, obj.o, []*pkcs11.Attribute{
 		&pkcs11.Attribute{Type: pkcs11.CKA_CLASS},
-		&pkcs11.Attribute{Type: pkcs11.CKA_KEY_TYPE},
 	})
 	if err != nil {
+		err = fmt.Errorf("Error reading CKA_CLASS: %x", err)
 		return
 	}
-
 	switch btoi(attrs[0].Value) {
 	case pkcs11.CKO_PUBLIC_KEY:
 		c = "public"
 	case pkcs11.CKO_PRIVATE_KEY:
 		c = "private"
+	case pkcs11.CKO_CERTIFICATE:
```

Comment:
As far as my experience has indicated, gnome keyring only supplies a certificate and a private key.